### PR TITLE
Looks/inbox redesign

### PR DESCRIPTION
**Pull request by jakerenzella**
# Description

This is a major PR towards migrating the frontend of Doubtfire.

* Upgrades to angular 10
* Migrates a number of tightly-coupled components
* Redesigns several components
* Switches to es2020 build files with IVY compiler
* Updates several libraries
* Adds frontend support for pinning tasks to inbox
* Adds support for keyboard navigation of task inbox (command + up/down)

### Discussion Comments
* Redesigns the Intelligent Discussion Comments, adding support to only playback responses
* Adds the ability to seek playback on all audio comments including discussion comments
* Adds a playback tone to signal when students should respond

### Design
* Implements a custom scrollbar style to maintain consistency across platforms when machines are in docked mode.
* Implements new design for task inbox, and comments panel.

Fixes # (issue)

* Fixes an issue where the comment read receipt did not update correctly on a refresh
* Fixes issues with the user icon component by recreating the component
* Fixes issues with mark by task definition causing errors requiring a refresh
* Fixes an issue with scrolling down the task comment panel

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Exploratory testing

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

If you have any questions, please contact @macite or @jakerenzella.

### CODE REVIEW

File: src/app/tasks/task-comments-viewer/task-comments-viewer.component.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+              <div *ngSwitchCase="'pdf'" class="comment-pdf">
+                <a (click)="openCommentsModal(comment)">
+                  <p><i class="fa fa-file-pdf-o"></i> view pdf</p>
+                </a>
+              </div>
+            </div>
+            <div *ngIf="isBubbleComment(comment.type)" class="comment-overflow">
+              <comment-bubble-action
+                ng-show="hover"
```
Comment (BaderAlotaibi) [inline src/app/tasks/task-comments-viewer/task-comments-viewer.component.html]: For this one , cant be [hidden]

Comment (jakerenzella) [reply src/app/tasks/task-comments-viewer/task-comments-viewer.component.html]: I'm not sure what you're referring to, this is outdated.

**Comment by macite** [comment]
Started a test run:

User Icon isn't centred in Welcome Screen.

image

Design looks great

Comment area jitters as you mouse over the comments.

OnTrackReview1

Long messages also expand comment area - not sure about this.

**Comment by macite** [comment]
Would be good to look at adding the icons to the drop down at some stage as well.

image

**Comment by macite** [comment]
Inbox is lost on Give Students Feedback page when size reduced.

image

**Comment by macite** [comment]
When updating a task status the task spacing in the inbox list the spacing looks wrong - selecting a different task and reselecting the updated task then looks "right"

image

**Comment by macite** [comment]
After recording and posting an audio comment, can we get the dialog to close? It remained open for me.

image

**Comment by macite** [comment]
Love it! ... soon ... :)

image

**Comment by jakerenzella** [comment]
> Love it! ... soon ... :)
> 
> image

To confirm, are you saying to remove it until it's ready? haha

**Comment by jakerenzella** [comment]
> Started a test run:
> 
> User Icon isn't centred in Welcome Screen.
> 
> image
> 
> Design looks great
> 
> Comment area jitters as you mouse over the comments.
> 
> OnTrackReview1
> 
> Long messages also expand comment area - not sure about this.



> Would be good to look at adding the icons to the drop down at some stage as well.
> 
> image

Don't want to fix this up, as it is yet to be migrated, will take a complete look at it then

**Comment by jakerenzella** [comment]
> Inbox is lost on Give Students Feedback page when size reduced. Also on Mark by Definition page.
> 
> image
> 
> Seems list was empty - but you cannot access the search component to select different task definitions etc.
> 
> Need something to "pop out" search

Good catch 👍

**Comment by jakerenzella** [comment]
> Started a test run:
> 
> User Icon isn't centred in Welcome Screen.
> 
> image
> 
> Design looks great
> 
> Comment area jitters as you mouse over the comments.
> 
> OnTrackReview1
> 
> Long messages also expand comment area - not sure about this.

I am not able to properly address the width and responsiveness of the overall page as the parent component is not migrated, this is because the PDF/task viewer component is yet to be migrated, so the approach I've taken here is a bit of a hack until I can apply the new flex layouts. So in short, I propose we accept it for now. Having said that I'll take a look to see if there are some ideas for a quick fix to some of these

**Comment by macite** [comment]
User icon looks a little small and hard to see

image

**Comment by jakerenzella** [comment]
> User icon looks a little small and hard to see
> 
> 
> 
> image
> 
> 

the one on the left? I can increase that one. The one on right is just the read receipt meant to be very small

**Comment by jakerenzella** [comment]
@macite I've addressed each issue, here's some quick demos!





Let me know if there are any further issues identified.

**Comment by macite** [comment]
Testing again and I have updated the user icon code (will push to your branch soon). There were issues when using a image rather than the text icons that are default. Please make sure to check this.

Also noticed an issue with the grade badge. It does not show for anything but the selected task, and not sure of the styling. We need to make this look more badge like. Also check the quality points badge. These both need to be visible so that you can scan the list and see these details.

image

**Comment by jakerenzella** [comment]
Yep, this is why I wanted to get this on UAT so I could test with various
types on inbox items, is that possible? My database isn’t that varied and
would like to see it with real data.

**Pull request merged**